src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -2101,6 +2101,15 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
                             break;
                     }
                 }
+                else if (highestBoundExpr is BoundConditionalOperator { WasTargetTyped: true } cond)
+                {
+                    type = cond.NaturalTypeOpt;
+                    convertedType = cond.Type;
```

Comment:
Nit: both of these could have been pulled out via declaration patterns.

Reply:
Yes, at the expense of two more declared variables that we do not need.

---
In reply to: [410425048](https://github.com/dotnet/roslyn/pull/40604#discussion_r410425048) [](ancestors = 410425048)